backend/Dockerfile

```diff
@@ -1,16 +1,34 @@
-FROM python:3.12.7
+FROM python:3.12.7-slim as base
 
-ENV PYTHONPATH "${PYTHONPATH}:/"
+ENV PYTHONPATH="${PYTHONPATH}:/"
 ENV PORT=8000
 
+# Install necessary dependencies
+RUN apt-get update && apt-get install -y curl
+
+# Install Poetry
+RUN curl -sSL https://install.python-poetry.org | POETRY_HOME=/opt/poetry python3 - && \
+    ln -s /opt/poetry/bin/poetry /usr/local/bin/poetry
+
 WORKDIR /app
 
-COPY pyproject.toml /app/
-COPY poetry.lock /app/
+COPY pyproject.toml poetry.lock /app/
 
-RUN pip install poetry
-RUN poetry install
+# Install dependencies using Poetry
+RUN poetry install --no-root
 
 COPY . .
 
+# Running tests
+FROM base as test
+RUN poetry run pytest
```

Comment:
keyword case not matching

Reply:
oh no

Reply:
well, it triggers warning during build
![image](https://github.com/user-attachments/assets/0dbe4f55-5015-4e99-8b2e-67a2076c47c1)